src/aiida/transports/plugins/ssh.py

```diff
@@ -1375,9 +1375,9 @@ def rename(self, oldpath: TransportPath, newpath: TransportPath):
         # why to raise an OSError if the newpath does not exist?
         # ofcourse newpath shouldn't exist, that's why we are renaming it!
         # issue opened here: https://github.com/aiidateam/aiida-core/issues/6725
-        if not self.isfile(newpath):
-            if not self.isdir(newpath):
-                raise OSError(f'Destination {newpath} does not exist')
+
+        if self.isfile(newpath) or self.isdir(newpath):
```

Comment:
```suggestion
        if self.path_exists(newpath):
            raise OSError(f'Destination {newpath} already exists')
```
maybe this is easier, and perhaps more readable.
Also I'm not entirely sure, if existing symlinks would get caught by `self.isfile`

Reply:
Please also get rid of the comments above, line 1374-1377